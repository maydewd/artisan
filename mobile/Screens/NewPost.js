/**
 * New Post Screen
 */

import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  TextInput,
  TouchableOpacity,
  Image,
  PixelRatio,
  Picker,
  Platform,
  ScrollView,
  AsyncStorage
} from 'react-native';
var NavigationBar = require('react-native-navbar');
import Button from 'react-native-button'
var ImagePicker = require('react-native-image-picker');
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import {getScreenWidth, getScreenHeight, usablePercent, usableWithTop} from '../helpers/dimension'
import {grabArtTypes} from '../resources/Types'
import { Kohana } from 'react-native-textinput-effects';
import ModalPicker from 'react-native-modal-picker';
import FA from 'react-native-vector-icons/FontAwesome';

import Geocoder from 'react-native-geocoder';
// TODO:
var MY_KEY = 'AIzaSyB4Wup3-phaP5kaiLHUOELxdtMKzm1GuxI';
Geocoder.fallbackToGoogle(MY_KEY);

class NewPost extends Component {

  constructor(props) {
    super(props);
    this.state = {
      description: '',
      price: null,
      photoSource: null,
      type: null,
      position: null,
      location: null,
    };
  }

  render() {
    var titleConfig = {
     title: 'New Post',
   };

   const leftButtonConfig = {
    title: 'Back',
    handler: () => {this.pop()}
  };

    return (
      <View>
        <NavigationBar
        style={styles.navBar}
        title={titleConfig}
        leftButton={leftButtonConfig}
        />
        <View style= {{height: usableWithTop()}}>
          <TouchableOpacity style = {styles.container} onPress={this.selectPhotoTapped.bind(this)}>
              <View>
                { this.state.photoSource === null ? <MaterialIcons name="add-a-photo" size= {50}/>:
                    <Image style={styles.avatar} source={this.state.photoSource} />
                  }
                </View>
            </TouchableOpacity>
            <Kohana
              style={{flex: 3, width: getScreenWidth(), backgroundColor: 'white', paddingTop: 10, borderColor: 'gray', borderTopWidth: 1}}
              onChangeText={(username) => this.setState({username})}
              label={'Story'}
              iconClass={FA}
              iconName={'book'}
              iconColor={'#24518D'}
              labelStyle={{ color: 'pink' }}
              inputStyle={{ color: '#24518D' }}
              multiline = {true}
              numberOfLines = {2}
              onChangeText={(description) => this.setState({description})}
            />
            <Kohana
              style={{flex: 1, width: getScreenWidth(), backgroundColor: 'white', borderColor: 'gray', borderTopWidth: 1}}
              onChangeText={(username) => this.setState({username})}
              label={'Asking Price'}
              iconClass={MaterialIcons}
              iconName={'payment'}
              iconColor={'#24518D'}
              labelStyle={{ color: 'pink' }}
              inputStyle={{ color: '#24518D' }}
              keyboardType = 'numeric'
              onChangeText={(price) => this.setState({price})}
            />
            <ModalPicker
                   data={grabArtTypes()}
                   style = {{flex:1, width: getScreenWidth()}}
                   initValue="Select a type"
                   onChange={(option)=> this.setState({type:option.label})}
                   >
                       <Kohana
                         style = {{width: getScreenWidth(), borderColor: 'gray', borderTopWidth: 1}}
                         editable= {false}
                         label={'Type'}
                         iconClass={MaterialIcons}
                         iconName={'subject'}
                         iconColor={'#24518D'}
                         labelStyle={{ color: 'pink' }}
                         inputStyle={{ color: '#24518D' }}
                         value={this.state.type}
                       />
            </ModalPicker>
            <Button
              containerStyle={{padding:10, overflow:'hidden', maxHeight: 50, backgroundColor: '#24518D', borderRadius: 2}}
              style={{fontSize: 20, color: 'white'}}
              onPress={() => this._postPressed()}>
              Post
            </Button>
        </View>
      </View>
    );
  }

  selectPhotoTapped() {
    console.log("Getting photo");
    const options = {
        quality: 1.0,
        maxWidth: 500,
        maxHeight: 500,
        storageOptions: {
          skipBackup: true
        }
      };
    ImagePicker.showImagePicker(options, (response) => {
        console.log('Response = ', response);

        if (response.didCancel) {
       console.log('User cancelled photo picker');
     }
     else if (response.error) {
       console.log('ImagePicker Error: ', response.error);
     }
     else if (response.customButton) {
       console.log('User tapped custom button: ', response.customButton);
     }
     else {
       var source;

       // You can display the image using either:
       //source = {uri: 'data:image/jpeg;base64,' + response.data, isStatic: true};

       //Or:
       if (Platform.OS === 'android') {
         source = {uri: response.uri, isStatic: true};
       } else {
         source = {uri: response.uri.replace('file://', ''), isStatic: true};
       }

       this.setState({
         photoSource: source
       });
     }
    });
  }
  _post() {
    const {description, type, price, photoSource, position, location} = this.state;
    console.log("description: " + description)
    console.log("type: " + type)
    console.log("photoSource: " + photoSource)
    console.log("position: " + JSON.stringify(position))
    console.log("location: " + location)
    console.log(AsyncStorage.getItem('jwtToken'));
    var request = new XMLHttpRequest();
    request.open("POST", "http://colab-sbx-137.oit.duke.edu:3000/api/listings");
    request.setRequestHeader('Accept', 'application/json');
    request.setRequestHeader('Content-Type', 'multipart/form-data');
    var body = new FormData();
    body.append('description', description);
    body.append('type', type);
    body.append('price', price);
    body.append('position', position);
    body.append('location', location);
    var photo;
    if (Platform.OS === 'android') {
      photo = {
        uri: photoSource.uri,
        type: 'image/jpeg',
        name: 'photo.jpg',
      };
    } else {
      photo = {
        uri: photoSource,
        type: 'image/jpeg',
        name: 'photo.jpg',
      };
    }
    body.append('image', photo);
    AsyncStorage.getItem('jwtToken', (err, result) => {
        request.setRequestHeader('Authorization', result);
        request.send(body);
        console.log(body);
        console.log(request);
        console.log('request sent');
        alert('Thanks!')
        this.pop()
    });
  }
  _getLocation() {

    navigator.geolocation.getCurrentPosition (
      (position) => {
        this.setState({position});
        var loc = {
          lat: position.coords.latitude,
          lng: position.coords.longitude
        }
        Geocoder.geocodePosition(loc).then(res => {
          console.log(res)
          // TODO: res is an array of geocoding objects, and the information is not guaranteed to be there
          this.setState({location: res[0].locality})
          this._post();
        })
        .catch(err => alert(JSON.stringify(err)))
      },
      (error) => alert(JSON.stringify(error)),
      {enableHighAccuracy: true, timeout: 2000, maximumAge: 1000}
    );
  }
  _postPressed() {
    this._getLocation();
  }
  pop() {
    this.props.navigator.pop()
  }

}

const styles = StyleSheet.create({
  navBar: {
    backgroundColor: '#cce5ff'
  },

  container: {
    width:  getScreenWidth(),
    height: usablePercent(50),
    alignItems: 'center',
    justifyContent: "space-around",
    backgroundColor: '#f6f6f6'
  },

  center: {
    alignItems: 'center',
    justifyContent: "space-around",
  },

  avatar: {
    resizeMode: 'contain',
    width: getScreenWidth(),
    height: usablePercent(50)
  },

  picker: {
    height: 150
  }
});


module.exports = NewPost
