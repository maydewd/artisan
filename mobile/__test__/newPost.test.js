import 'react-native';
import React from 'react';
import NewPost from '../Screens/NewPost.js';

import renderer from 'react-test-renderer';

it('New post renders correctly', () => {
  const tree = renderer.create(
    <NewPost />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
