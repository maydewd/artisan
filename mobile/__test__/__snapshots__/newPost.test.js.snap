exports[`test New post renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
        },
        null,
      ]
    }>
    <View
      style={
        Array [
          Object {
            "height": 20,
          },
          null,
        ]
      } />
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "space-between",
          },
          Object {
            "backgroundColor": "#cce5ff",
          },
        ]
      }>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#333",
                "fontSize": 17,
                "fontWeight": "500",
                "letterSpacing": 0.5,
              },
              null,
            ]
          }>
          New Post
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "stretch",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID={undefined}>
          <View
            style={
              Array [
                undefined,
                Object {
                  "marginLeft": 8,
                },
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 17,
                    "letterSpacing": 0.5,
                  },
                  Object {
                    "color": "#0076FF",
                  },
                ]
              }>
              Back
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "stretch",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID={undefined}>
          <View
            style={
              Array [
                undefined,
                Object {
                  "marginRight": 8,
                },
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 17,
                    "letterSpacing": 0.5,
                  },
                  Object {
                    "color": "#0076FF",
                  },
                ]
              }>
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    keyboardVerticalOffset={0}
    onLayout={[Function]}
    style={undefined}>
    <View
      style={
        Array [
          undefined,
          Object {
            "bottom": 0,
          },
        ]
      }>
      <View
        style={
          Object {
            "height": 602,
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f6f6f6",
              "height": 276,
              "justifyContent": "space-around",
              "opacity": 1,
              "width": 375,
            }
          }
          testID={undefined}>
          <View>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontFamily": "Material Icons",
                    "fontSize": 50,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  undefined,
                ]
              }>
              
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "white",
                "borderColor": "gray",
                "borderTopWidth": 1,
                "flex": 3,
                "paddingTop": 10,
                "width": 375,
              },
            ]
          }>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "justifyContent": "center",
                "marginLeft": -40,
                "padding": 16,
              }
            }
            testID={undefined}>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#24518D",
                    "fontFamily": "FontAwesome",
                    "fontSize": 25,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  undefined,
                ]
              }>
              
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "left": 16,
                "opacity": 1,
                "position": "absolute",
                "top": 16,
              }
            }
            testID={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#D2D2D2",
                    "fontFamily": "Arial",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  },
                  Object {
                    "color": "pink",
                  },
                ]
              }>
              Story
            </Text>
          </View>
          <TextInput
            easing={[Function]}
            iconClass={[Function]}
            iconColor="#24518D"
            iconName="book"
            inputStyle={
              Object {
                "color": "#24518D",
              }
            }
            label="Story"
            labelStyle={
              Object {
                "color": "pink",
              }
            }
            multiline={true}
            numberOfLines={2}
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            style={
              Array [
                Object {
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "paddingHorizontal": 16,
                  "paddingVertical": 0,
                },
                Object {
                  "color": "#24518D",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value={undefined} />
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "white",
                "borderColor": "gray",
                "borderTopWidth": 1,
                "flex": 1,
                "width": 375,
              },
            ]
          }>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "justifyContent": "center",
                "marginLeft": -40,
                "padding": 16,
              }
            }
            testID={undefined}>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#24518D",
                    "fontFamily": "Material Icons",
                    "fontSize": 25,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  undefined,
                ]
              }>
              
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "left": 16,
                "opacity": 1,
                "position": "absolute",
                "top": 16,
              }
            }
            testID={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#D2D2D2",
                    "fontFamily": "Arial",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  },
                  Object {
                    "color": "pink",
                  },
                ]
              }>
              Asking Price
            </Text>
          </View>
          <TextInput
            easing={[Function]}
            iconClass={[Function]}
            iconColor="#24518D"
            iconName="payment"
            inputStyle={
              Object {
                "color": "#24518D",
              }
            }
            keyboardType="numeric"
            label="Asking Price"
            labelStyle={
              Object {
                "color": "pink",
              }
            }
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            style={
              Array [
                Object {
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "paddingHorizontal": 16,
                  "paddingVertical": 0,
                },
                Object {
                  "color": "#24518D",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value={undefined} />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "width": 375,
            }
          }>
          <Modal
            animationType="slide"
            onRequestClose={[Function]}
            transparent={true}
            visible={false}>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(0,0,0,0.7)",
                    "height": 667,
                    "width": 375,
                  },
                  Object {},
                ]
              }>
              <View
                style={
                  Object {
                    "backgroundColor": "rgba(255,255,255,0.8)",
                    "borderRadius": 5,
                    "height": 400,
                    "left": 37.5,
                    "top": 133.5,
                    "width": 300,
                  }
                }>
                <ScrollView
                  keyboardShouldPersistTaps={true}>
                  <View
                    style={
                      Object {
                        "paddingHorizontal": 10,
                      }
                    }>
                    <View
                      style={
                        Array [
                          Object {
                            "borderBottomColor": "#ccc",
                            "borderBottomWidth": 1,
                            "padding": 16,
                          },
                          Object {},
                        ]
                      }>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "fontSize": 16,
                              "textAlign": "center",
                            },
                            Object {},
                          ]
                        }>
                        Types
                      </Text>
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hitSlop={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID={undefined}>
                      <View
                        style={
                          Array [
                            Object {
                              "borderBottomColor": "#ccc",
                              "borderBottomWidth": 1,
                              "padding": 8,
                            },
                            Object {},
                          ]
                        }>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "rgba(0,118,255,0.9)",
                                "fontSize": 16,
                                "textAlign": "center",
                              },
                              Object {},
                            ]
                          }>
                          Ceramics
                        </Text>
                      </View>
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hitSlop={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID={undefined}>
                      <View
                        style={
                          Array [
                            Object {
                              "borderBottomColor": "#ccc",
                              "borderBottomWidth": 1,
                              "padding": 8,
                            },
                            Object {},
                          ]
                        }>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "rgba(0,118,255,0.9)",
                                "fontSize": 16,
                                "textAlign": "center",
                              },
                              Object {},
                            ]
                          }>
                          Painting
                        </Text>
                      </View>
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hitSlop={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID={undefined}>
                      <View
                        style={
                          Array [
                            Object {
                              "borderBottomColor": "#ccc",
                              "borderBottomWidth": 1,
                              "padding": 8,
                            },
                            Object {},
                          ]
                        }>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "rgba(0,118,255,0.9)",
                                "fontSize": 16,
                                "textAlign": "center",
                              },
                              Object {},
                            ]
                          }>
                          Photograph
                        </Text>
                      </View>
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hitSlop={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID={undefined}>
                      <View
                        style={
                          Array [
                            Object {
                              "borderBottomColor": "#ccc",
                              "borderBottomWidth": 1,
                              "padding": 8,
                            },
                            Object {},
                          ]
                        }>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "rgba(0,118,255,0.9)",
                                "fontSize": 16,
                                "textAlign": "center",
                              },
                              Object {},
                            ]
                          }>
                          Drawing
                        </Text>
                      </View>
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hitSlop={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID={undefined}>
                      <View
                        style={
                          Array [
                            Object {
                              "borderBottomColor": "#ccc",
                              "borderBottomWidth": 1,
                              "padding": 8,
                            },
                            Object {},
                          ]
                        }>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "rgba(0,118,255,0.9)",
                                "fontSize": 16,
                                "textAlign": "center",
                              },
                              Object {},
                            ]
                          }>
                          Jewelry
                        </Text>
                      </View>
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hitSlop={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID={undefined}>
                      <View
                        style={
                          Array [
                            Object {
                              "borderBottomColor": "#ccc",
                              "borderBottomWidth": 1,
                              "padding": 8,
                            },
                            Object {},
                          ]
                        }>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "rgba(0,118,255,0.9)",
                                "fontSize": 16,
                                "textAlign": "center",
                              },
                              Object {},
                            ]
                          }>
                          Sculpture
                        </Text>
                      </View>
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hitSlop={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID={undefined}>
                      <View
                        style={
                          Array [
                            Object {
                              "borderBottomColor": "#ccc",
                              "borderBottomWidth": 1,
                              "padding": 8,
                            },
                            Object {},
                          ]
                        }>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "rgba(0,118,255,0.9)",
                                "fontSize": 16,
                                "textAlign": "center",
                              },
                              Object {},
                            ]
                          }>
                          Metal Work
                        </Text>
                      </View>
                    </View>
                  </View>
                </ScrollView>
              </View>
              <View
                style={
                  Object {
                    "left": 37.5,
                    "top": 143.5,
                  }
                }>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}>
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "rgba(255,255,255,0.8)",
                          "borderRadius": 5,
                          "padding": 8,
                          "width": 300,
                        },
                        Object {},
                      ]
                    }>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#333",
                            "fontSize": 16,
                            "textAlign": "center",
                          },
                          Object {},
                        ]
                      }>
                      cancel
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </Modal>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "white",
                    "flex": 1,
                    "flexDirection": "row",
                    "overflow": "hidden",
                  },
                  Object {
                    "borderColor": "gray",
                    "borderTopWidth": 1,
                    "width": 375,
                  },
                ]
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "justifyContent": "center",
                    "marginLeft": -40,
                    "padding": 16,
                  }
                }
                testID={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#24518D",
                        "fontFamily": "Material Icons",
                        "fontSize": 25,
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      undefined,
                    ]
                  }>
                  
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "left": 16,
                    "opacity": 1,
                    "position": "absolute",
                    "top": 16,
                  }
                }
                testID={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#D2D2D2",
                        "fontFamily": "Arial",
                        "fontSize": 18,
                        "fontWeight": "bold",
                      },
                      Object {
                        "color": "pink",
                      },
                    ]
                  }>
                  Type
                </Text>
              </View>
              <TextInput
                easing={[Function]}
                editable={false}
                iconClass={[Function]}
                iconColor="#24518D"
                iconName="subject"
                inputStyle={
                  Object {
                    "color": "#24518D",
                  }
                }
                label="Type"
                labelStyle={
                  Object {
                    "color": "pink",
                  }
                }
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                style={
                  Array [
                    Object {
                      "color": "black",
                      "flex": 1,
                      "fontSize": 18,
                      "paddingHorizontal": 16,
                      "paddingVertical": 0,
                    },
                    Object {
                      "color": "#24518D",
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value={undefined} />
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#24518D",
              "borderRadius": 2,
              "maxHeight": 50,
              "opacity": 1,
              "overflow": "hidden",
              "padding": 10,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontFamily": "HelveticaNeue-Medium",
                  "fontSize": 17,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                null,
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                null,
              ]
            }>
            Post
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
